odrive-motor/src/odriveCAN.py

```diff
@@ -0,0 +1,290 @@
+        return Motor.Properties(position_reporting=True)
+    
+    async def stop(self, extra: Optional[Dict[str, Any]] = None, **kwargs):
+        msg = db.get_message_by_name('Set_Axis_State')
+        data = msg.encode({'Axis_Requested_State': 0x01})
+        msg = can.Message(arbitration_id=msg.frame_id | self.nodeID << 5, is_extended_id=False, data=data)
+        await self.try_send(msg)
+
+    async def is_powered(self, extra: Optional[Dict[str, Any]] = None, **kwargs) -> Tuple[bool, float]:
+        current_power = 0
+        for msg in bus:
+            if msg.arbitration_id == ((self.nodeID << 5) | db.get_message_by_name('Heartbeat').frame_id):
+                current_state = db.decode_message('Heartbeat', msg.data)['Axis_State']
+                if (current_state != 0x0) & (current_state != 0x1):
+                    msg = db.get_message_by_name('Get_Iq')
+                    data = msg.encode({'Iq_Setpoint': 0, 'Iq_Measured': 0})
+                    msg = can.Message(arbitration_id=0x014 | self.nodeID << 5, is_extended_id=False, data=data)
+                    self.try_send(msg)
+
+                    for msg1 in bus:
+                        if msg1.arbitration_id == ((self.nodeID << 5) | 0x014):
+                            current = db.decode_message('Get_Iq', msg1.data)['Iq_Setpoint']
+                            current_power = current/self.current_soft_max
+                            return [True, current_power]
+                else:
+                    return [False, 0]
+
+    
+    async def is_moving(self) -> bool:
+        for msg in bus:
+            if msg.arbitration_id == ((self.nodeID << 5) | db.get_message_by_name('Heartbeat').frame_id):
+                current_state = db.decode_message('Heartbeat', msg.data)['Axis_State']
+                if (current_state != 0x0) & (current_state != 0x1):
+                    return True
+                else:
+                    return False
+                
+    async def do_command(self) -> Dict[str, Any]:
+        pass
+
+    async def wait_until_correct_state(self, state):
+        for msg in bus:
+            current_state = db.decode_message('Heartbeat', msg.data)['Axis_State']
+            if current_state == state:
+                return
+   
+    async def wait_and_set_to_idle(self, goal):
+        while True:
+            position = await self.get_position()
+            if abs(position - goal) <= 0.01:
+                msg = db.get_message_by_name('Set_Axis_State')
+                data = msg.encode({'Axis_Requested_State': 0x01})
+                msg = can.Message(arbitration_id=msg.frame_id | self.nodeID << 5, is_extended_id=False, data=data)
+                await self.try_send(msg)
+
+    async def try_send(self, msg):
+        try:
+            bus.send(msg)
+        except can.CanError:
+            LOGGER.error("Message NOT sent! Please verify can0 is working first")
+
+    async def surface_errors(self):
+        for msg in bus:
```

Comment:
you have couple of places where you get can messages from the bus concurrently is there a risk each of this call consumes and discards messages not meant for them?